java-client/src/main/java/org/apache/pegasus/client/PegasusScanner.java

```diff
@@ -88,9 +88,27 @@ public PegasusScanner(
     _needCheckHash = needCheckHash;
     _incomplete = false;
     _fullScan = fullScan;
+    _nextItem = null;
```

Comment:
The global variable will cause `thread-safe` problem,  in origin code, I remember it will `lock` before fetch `next`: https://github.com/WHBANG/incubator-pegasus/blob/bd099a9c963de8de4cd8ad4c80ca6bb9653d24cd/java-client/src/main/java/org/apache/pegasus/client/PegasusScanner.java#L125

Please check and make sure the `refactor` is `thread-safe`

Reply:
OK, i will add  _nextItemLock to make _nextItem thread-safe